src/Pages/ReservationsPage/ReservationsPage.js

```diff
@@ -1,22 +1,950 @@
 // Import Components 
-import React from 'react';
-import GeneralPage from '../GeneralPage/GeneralPage';
+import React, { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { API } from '../../Constants';
+import { connect } from 'react-redux';
+import { resetUserData } from '../../storage';
 
 // Import Stylings
 import './ReservationsPage.css';
-import { HiExclamation } from 'react-icons/hi';
+
+//#region Import UI Components
+import GeneralPage from '../GeneralPage/GeneralPage';
+import Logo from '../../Components/Logo/Logo';
+import SearchBarInputField from '../../Components/InputFields/SearchBarInputField';
+import DatePickerInputField from '../../Components/InputFields/DatePickerInputField/DatePickerInputField';
+import StandardButton from '../../Components/Buttons/StandardButton/StandardButton';
+import AvailableModelList from '../../Components/Lists/AvailableModelList/AvailableModelList';
+import UnauthorizedPanel from '../../Components/Panels/UnauthorizedPanel/UnauthorizedPanel';
+import IconButton from '../../Components/Buttons/IconButton/IconButton';
+import SpecifyModelReservationQuantityList from '../../Components/Lists/SpecifyModelReservationQuatityList/SpecifyModelReservationQuantityList';
+import ReservationConfirmationDetailsList from '../../Components/Lists/ReservationConfirmationDetailsList/ReservationConfirmationDetailsList';
+import IconModal from '../../Components/Modals/IconModal/IconModal';
+import FilterButton from '../../Components/Buttons/FilterButton/FilterButton';
+import ReservationList from '../../Components/Lists/ReservationList/ReservationList';
+import DetailSection from '../../Components/Sections/DetailSection/DetailSection';
+//#endregion
+
+//#region Import Icons
+import { HiArrowSmRight, HiCalendar, HiCheck, HiChevronLeft, HiChevronRight, HiExclamationCircle, HiMinusCircle, HiPencilAlt, HiPlus, HiRefresh, HiX } from 'react-icons/hi';
+import { MdCheckBoxOutlineBlank, MdCheckBox } from 'react-icons/md';
+//#endregion
 
 // Define ReservationsPage Component
-function ReservationsPage() {
+function ReservationsPage(props) {
```

Comment:
This is a massive function! I don't know much about React and all that, but I wonder if it is possible to split this into some helpers?

Reply:
Yeah, it actually is a massive function for handling three user roles. It is quite complicated to split into other components further due to the dependent states.

Reply:
Got it 👍